pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,831 @@
+	}
+
+	pub(super) fn allowance(
+		session: &mut Session<Sandbox>,
+		owner: AccountId32,
+		spender: AccountId32,
+	) -> Balance {
+		call::<Balance>(
+			session,
+			"PSP22::allowance",
+			vec![owner.to_string(), spender.to_string()],
+			None,
+		)
+		.unwrap()
+	}
+
+	pub(super) fn transfer(
+		session: &mut Session<Sandbox>,
+		to: AccountId32,
+		amount: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22::transfer",
+			vec![
+				to.to_string(),
+				amount.to_string(),
+				serde_json::to_string::<[u8; 0]>(&[]).unwrap(),
+			],
+			None,
+		)
+	}
+
+	pub(super) fn transfer_from(
+		session: &mut Session<Sandbox>,
+		from: AccountId32,
+		to: AccountId32,
+		amount: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22::transfer_from",
+			vec![
+				from.to_string(),
+				to.to_string(),
+				amount.to_string(),
+				serde_json::to_string::<[u8; 0]>(&[]).unwrap(),
+			],
+			None,
+		)
+	}
+
+	pub(super) fn approve(
+		session: &mut Session<Sandbox>,
+		spender: AccountId32,
+		value: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(session, "PSP22::approve", vec![spender.to_string(), value.to_string()], None)
+	}
+
+	pub(super) fn increase_allowance(
+		session: &mut Session<Sandbox>,
+		spender: AccountId32,
+		value: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22::increase_allowance",
+			vec![spender.to_string(), value.to_string()],
+			None,
+		)
+	}
+
+	pub(super) fn decrease_allowance(
+		session: &mut Session<Sandbox>,
+		spender: AccountId32,
+		value: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22::decrease_allowance",
+			vec![spender.to_string(), value.to_string()],
+			None,
+		)
+	}
+
+	/// PSP22Metadata test utilities.
+
+	pub(super) fn token_name(session: &mut Session<Sandbox>) -> Option<String> {
+		call::<Option<String>>(session, "PSP22Metadata::token_name", vec![], None).unwrap()
+	}
+
+	pub(super) fn token_symbol(session: &mut Session<Sandbox>) -> Option<String> {
+		call::<Option<String>>(session, "PSP22Metadata::token_symbol", vec![], None).unwrap()
+	}
+
+	pub(super) fn token_decimals(session: &mut Session<Sandbox>) -> u8 {
+		call::<u8>(session, "PSP22Metadata::token_decimals", vec![], None).unwrap()
+	}
+
+	/// PSP22Mintable test utilities.
+
+	pub(super) fn mint(
+		session: &mut Session<Sandbox>,
+		account: AccountId32,
+		amount: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22Mintable::mint",
+			vec![account.to_string(), amount.to_string()],
+			None,
+		)
+	}
+
+	/// PSP22Burnable test utilities.
+
+	pub(super) fn burn(
+		session: &mut Session<Sandbox>,
+		account: AccountId32,
+		amount: Balance,
+	) -> Result<(), PSP22Error> {
+		call::<()>(
+			session,
+			"PSP22Burnable::burn",
+			vec![account.to_string(), amount.to_string()],
+			None,
+		)
+	}
+
+	/// Call a contract method and decode the returned data.
+	pub(super) fn call<T: Decode>(
```

Comment:
If possible, should be moved to pop-drink as generic functions taking account id and error.  Defining this in every example, or contract, adds complexity and developer burden.

Reply:
Done